File: src/Media.java
/**
 * Media class is a super class of Movie, Music, TVSeries.
 * 
 * @author      Juha Hirvasniemi <jhirvasn@paju.oulu.fi>, Tapio Korvala <korvatap@paju.oulu.fi>
 * @version     1.0
 * @since       2013-12-30
 */
public class Media implements MediaItem {

	static final long serialVersionUID = 53L;
	private String title;
	private int publishYear;
	private int rating;
	private String genre;
	private static int id;
	
	/**
	 * Constuctor for class media.
	 */
	public Media() {
		increaseId();
	}
	
	/**
	 * Another constructor for class media.
	 * Can set a title for the object.
	 * 
	 * @param title		title for Media object.
	 */
	public Media(String title){
        this.title = title;
    }
	
	/**
	 * Returns the id of the Media Object.
	 * 
	 * @return id of the object.
	 */
	public int getId() {
		return id;
	}
	
	/**
	 * Used to increase id by one.
	 */
	public void increaseId() {
		id++;
	}
	
	/**
	 * Used to get the current maximum id.
	 * 
	 * @return current max id.
	 */
	public int getCurrentMaxId() {
		return getId();
	}
	
	/**
	 * Used to set the title of the Media.
	 * 
	 * @param title title to be set.
	 */
	public void setTitle(String title) {
		this.title = title;
	}
	
	/**
	 * Returns the title of the Media.
	 * 
	 * @return title of the media.
	 */
	public String getTitle() {
		return this.title;
	}
	
	/**
	 * Used to se the publish year of the media.
	 * 
	 * @param publishYear year to be set.
	 */
	public void setPublishYear(int publishYear) {
		this.publishYear = publishYear;
	}
	
	/**
	 * Returns the publish year of the Media.
	 * 
	 * @return publishYear of the media.
	 */
	public int getPublishYear() {
		return this.publishYear;
	}
	
	/**
	 * Used to set the rating of the Media.
	 * 
	 * @param rating value to be set.
	 */
	public void setRating(int rating) {
		this.rating = rating;
	}
	
	/**
	 * Returns the rating of the Media.
	 * 
	 * @return rating of the media.
	 */
	public int getRating() {
		return this.rating;
	}
	
	/**
	 * Used to set the genre of the Media.
	 * 
	 * @param genre value to be set.
	 */
	public void setGenre(String genre) {
		this.genre = genre;
	}
	
	/**
	 * Returns the genre of the Media.
	 * 
	 * @return genre of the media.
	 */
	public String getGenre() {
		return genre;
	}

	/**
	 * Inherited function not used in Media.
	 * @see MediaObject
	 */
	@Override
	public void add(MediaItem item) {
	}

	/**
	 * Inherited function not used in Media.
	 * @see MediaObject
	 */
	@Override
	public void print() {
	}

	/**
	 * Inherited function not used in Media.
	 * @see Movie
	 * @see Music
	 * @see TVSeries
	 */
	@Override
	public String[] getRow() {
		return null;
	}

	/**
	 * Inherited function not used in Media.
	 * @see MediaObject
	 */
	@Override
	public void deleteObjectById(int id) {
	}
	
	/**
	 * Used to set the id of Media.
	 * 
	 * @param id	id to be set.
	 */
	public static void setId(int id) {
		Media.id = id;
	}

}

File: web/Media.php
<?php

/**
 * Media class is a base class of Movie, Music and TVSeries.
 * 
 * @author      Juha Hirvasniemi <jhirvasn@paju.oulu.fi>, Tapio Korvala <korvatap@paju.oulu.fi>
 * @version     1.0
 * @since       2013-12-30
 */
class Media implements MediaItem {

    private $title;
    private $publishYear;
    private $rating;
    private $genre;
    static private $id = 0;
    
    public function __construct() {
        $arguments = func_get_args();
        $numOfArgs = func_num_args();
        if (method_exists ($this, $function = '__construct' .$numOfArgs) ) {
            call_user_func_array (array ($this, $function), $arguments);
        } 
    }

    /**
     * Constructor for class Media.
     * Sets the title for the Media object.
     * 
     * @param title     title for the Media object.
     */
    public function __construct1($title) {
            $this->title = $title;
    }

    /**
     * Used to set the title of the Media.
     * 
     * @param title     title to be set.
     */
    public function setTitle($title) {
        $this->title = $title;
    }
    
    /**
     * Returns the title of the Media.
     * 
     * @return      title of the media.
     */
    public function getTitle() {
        return $this->title;
    }
    
    /**
     * Sets the publish year of the media.
     *
     * @param publishYear   year to be set.
     */
    public function setPublishYear($publishYear) {
        $this->publishYear = $publishYear;
    }
    
    /**
     * Returns the publish year of the Media.
     * 
     * @return      publishYear of the media.
     */
    public function getPublishYear() {
        return $this->publishYear;
    }
    
    /**
     * Sets the rating of the Media.
     *
     * @param rating    value to be set.
     */
    public function setRating($rating) {
        $this->rating = $rating;
    }
    
    /**
     * Returns the rating of the Media.
     * 
     * @return      rating of the media.
     */
    public function getRating() {
        return $this->rating;
    }
    
    /**
     * Sets the genre of the Media.
     * 
     * @param genre     value to be set as genre.
     */
    public function setGenre($genre) {
        $this->genre = $genre;
    }
    
    /**
     * Returns the genre of the Media.
     * 
     * @return      genre of the media.
     */
    public function getGenre() {
        return $this->genre;
    }
    
    /**
     * Function not used in Media.
     * @see MediaObject
     */
    public function add($item) {
    }
    
    /**
     * Function not used in Media.
     * @see MediaObject
     */
    public function info() {
    }
    
    /**
     * Returns the id for a new object instantiated from
     * Media classes subclass.
     * 
     * @return      id for a new object.
     */
    static public function getId() {
        return self::$id;
    }
    
    /**
     * Sets the static id.
     * 
     * @param id    id to be set.
     */
    static public function setId($id) {
        self::$id = $id;
    }
    
    /**
     * Function that is not used in Media.
     */
    public function id() {
    }
    
    /**
     * Increases static id counter by one.
     */
    static protected function increaseId() {
        self::$id++;
    }
}
?>